crates/bevy_ecs/src/system/mod.rs

```diff
@@ -138,18 +124,109 @@ mod tests {
         world.spawn().insert(A);
 
         system.initialize(&mut world);
-        for archetype in world.archetypes.iter() {
-            system.new_archetype(archetype);
-        }
         system.run((), &mut world);
     }
 
-    fn run_system<Param, S: IntoSystem<(), (), Param>>(world: &mut World, system: S) {
-        let mut schedule = Schedule::default();
-        let mut update = SystemStage::parallel();
-        update.add_system(system);
-        schedule.add_stage("update", update);
-        schedule.run(world);
+    #[test]
+    fn exclusive_system() {
+        fn sys(world: &mut World) -> usize {
+            world.spawn().insert(A);
+
+            let mut count = 0;
+            for _ in world.query::<&A>().iter(world) {
+                count += 1;
+            }
+
+            count
+        }
+
+        let mut system = IntoSystem::into_system(sys);
+        let mut world = World::new();
+
+        system.initialize(&mut world);
+        let count = system.run((), &mut world);
+        assert_eq!(count, 1);
+    }
+
+    #[test]
+    #[should_panic]
+    fn crazy_dance1() {
```

Comment:
Drop a comment before this to explain the context of these tests please :)

Reply:
Or you know, we can find better names.

Reply:
I can do both. I'm probably also going move these soundness tests to their own file.

Reply:
Yes please. More dedicated test files in bevy_ecs please.